inference-engine/src/multi_device/multi_device_exec_network.cpp

```diff
@@ -380,6 +380,16 @@ InferenceEngine::IInferRequestInternal::Ptr MultiDeviceExecutableNetwork::Create
     InferenceEngine::SoIInferRequestInternal request_to_share_blobs_with;
 
     if (_workModeIsAUTO) {
+        if (!_networkFirstReady && _networkActualNeeded) {
```

Comment:
this (and the "if" before) are the only diff vs original loop below.

Reply:
Hi Maxim, thanks for the clear explanation.
1) shall we need to handle the context for different hw like GPU/VPU, they have separate context related, openCL implementation for GPU which may involve opencl dependencies as you mentioned, and for vpu, the VPU context. 
2) with make_shared_blob(desc, context) in the future, does the context need to be shared between loadNetwork to the hw device and make_shared_blob?

Reply:
>>with make_shared_blob(desc, context) in the future, does the context need to be shared between loadNetwork to the hw device and make_shared_blob?

Yes, and tis is why the LoadNetwork flavor that accepts conetxt will be likely needed